R/sampleSummaryStats.R

```diff
@@ -1,134 +1,140 @@
-#' @export
-setGeneric("sampleSummaryStatsTable", function(inSCE, ...) standardGeneric("sampleSummaryStatsTable"))
-
-#' @export
-setGeneric("sampleSummaryStatsTable<-", function(inSCE, ..., value) standardGeneric("sampleSummaryStatsTable<-"))
-
-#' @export
+#' @rdname sampleSummaryStatsTable
 setMethod("sampleSummaryStatsTable", "SingleCellExperiment", function(inSCE, statsName, ...){
-    return(inSCE@metadata$sctk$sampleSummary[[statsName]])
+    allStatsNames <- listSampleSummaryStatsTables(inSCE)
+    if(length(allStatsNames) == 0){
```

Comment:
If the error checking is done in listSampleSummaryStats, then it probably doesn't need to be done again here. This will allow you to have a single set of error statements